runner_service/inventory.py

```diff
@@ -91,37 +99,52 @@ def load(self):
 
         if not os.path.exists(self.filename):
             try:
-                with open(self.filename, 'w') as inv:
+                # Using Python 3 exclusive creation
+                with open(self.filename, 'x') as inv:                    
                     inv.write(yaml.dump(AnsibleInventory.inventory_seed,
-                                        default_flow_style=False))
+                                            default_flow_style=False))
+            except FileExistsError:
```

Comment:
FileExistsError....but you're inside a clause that says the file doesn't exists?

Reply:
But in a system with concurrency ... other instance of the service can be doing the same at the same time .... is a race condition

Reply:
Agree - it wouldn't be another service, but it could feasibly be another request